miden-lib/asm/sat/internal/asset_vault.masm

```diff
@@ -92,32 +88,33 @@ export.add_fungible_asset
     dup.4 movdn.5 push.0 movdn.3 dup movdn.4
     # => [ASSET_KEY, faucet_id, amount, vault_root_ptr, vault_root_ptr]
 
-    # get the asset vault root and read asset
-    padw movup.10 mem_loadw swapw exec.smt::get drop movup.7
-    # => [CUR_ASSET, VAULT_ROOT, amount, vault_root_ptr]
+    # get the asset vault root and read the vault asset value using the `push_smtpeek` decorator
+    padw movup.10 mem_loadw swapw adv.push_smtpeek adv_loadw swapw dupw.1 drop movup.11
+    # => [CUR_ASSET, VAULT_ROOT, CUR_VAULT_VALUE, amount, vault_root_ptr]
```

Comment:
`CUR_ASSET` and `CUR_VAULT_VALUE` are the same, right? If so, we may be able to avoid manipulations with the `faucet_id` and would not need to make a duplicate of it in the previous step.

Reply:
Not quite.  There is an edge case when the `CUR_VAULT_VALUE` is an empty word.  As such we drop the first element and replace it with the faucet_id such that it represents an asset with 0 balance.

Reply:
Ah - nice! I didn't think about this. Let's add a small comment explaining this?